ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUtils.java

```diff
@@ -316,12 +319,11 @@ private static Statistics collectStatistics(HiveConf conf, PrunedPartitionList p
 
       basicStatsFactory.addEnhancer(new BasicStats.RowNumEstimator(estimateRowSizeFromSchema(conf, schema)));
 
-      List<BasicStats> partStats = new ArrayList<>();
+      List<BasicStats> partStats =
+          partList.getNotDeniedPartns().parallelStream().
```

Comment:
Wouldn't this use the common pool for the JVM and could have issues when concurrent queries are running? May be better to provide local ForkJoinPool instead.

Reply:
@rbalamohan It seems parallel stream by default uses ForkJoinPool (citing Java in Action book)

Reply:
The point is `basicStatsFactory.build(Partish.buildFor(table, p))` can block the common pool shared with a whole JVM since it executes blocking-IO?